src/gui/sharemanager.cpp

```diff
@@ -255,10 +256,20 @@ QString LinkShare::getToken() const
 
 void LinkShare::setExpireDate(const QDate &date)
 {
+    shareJob(&LinkShare::slotExpireDateSet)->setExpireDate(getId(), date);
+}
+
+void LinkShare::setLabel(const QString &label)
+{
+    shareJob(&LinkShare::slotLabelSet)->setLabel(getId(), label);
```

Comment:
The template function is nice :) maybe we could change the name of the function to something like `createShareJob()`. Would it make at least for me more obvious.